internal/encryption/method/aesgcm/config.go

```diff
@@ -8,18 +8,124 @@ package aesgcm
 import (
 	"fmt"
 
+	"github.com/opentofu/opentofu/internal/collections"
+
 	"github.com/opentofu/opentofu/internal/encryption/method"
 )
 
+// The following settings follow the NIST SP 800-38D recommendation.
+const (
+	defaultTagSize   = 16
+	defaultNonceSize = 12
+	minimumNonceSize = 1
+)
+
+// validKeyLengths holds the valid key lengths supported by this method.
+var validKeyLengths = collections.Set[int]{
+	16: {},
+	24: {},
+	32: {},
+}
+
+// validTagSizes contains the valid tag sizes according to NIST SP 800-38D 5.2.1.2
+var validTagSizes = collections.Set[int]{
+	// These values are not supported by Go, see gcmMinimumTagSize in aes_gcm.go. They should also not be used for
+	// general-purpose applications.
+	// 4:  {},
+	// 8:  {},
+	12: {},
+	13: {},
+	14: {},
+	15: {},
+	16: {},
+}
+
+// Config is the configuration for the AES-GCM method.
 type Config struct {
-	Key []byte `hcl:"cipher"`
+	// Key is the encryption key for the AES-GCM encryption. It has to be 16, 24, or 32 bytes long for AES-128, 192, or
+	// 256, respectively.
+	Key []byte `hcl:"key" json:"key" yaml:"key"`
+
+	// AAD is the Additional Authenticated Data that is authenticated, but not encrypted. In the Go implementation, this
+	// data serves as a canary value against replay attacks. The AAD value on decryption must match this setting,
+	// otherwise the decryption will fail. (Note: this is Go-specific and differs from the NIST SP 800-38D description
+	// of the AAD.)
+	AAD []byte `hcl:"aad,optional" json:"aad,omitempty" yaml:"aad,omitempty"`
+
+	// NonceSize describes the length of the nonce. The default (and minimum) value is 12 bytes as per the NIST
```

Comment:
If you were to make anything configurable, I would opt for different named schemes: call this one `aes-gcm`  with default 12-bit nonce and 16-bit tag and introduce others later (`xchacha20-poly1305` or `aes-gcm-siv`). This gives you extensibility in the direction that matters, and ensures users don't shoot themselves in the foot.